src/components/Web3Provider/index.tsx

```diff
@@ -1,15 +1,27 @@
 import { useWeb3React, Web3ReactHooks, Web3ReactProvider } from '@web3-react/core'
 import { Connector } from '@web3-react/types'
 import { Connection } from 'connection'
+import { ConnectionType, setErrorHandler } from 'connection'
 import { getConnectionName } from 'connection/utils'
 import { isSupportedChain } from 'constants/chains'
 import { RPC_PROVIDERS } from 'constants/providers'
 import { TraceJsonRpcVariant, useTraceJsonRpcFlag } from 'featureFlags/flags/traceJsonRpc'
 import useEagerlyConnect from 'hooks/useEagerlyConnect'
 import useOrderedConnections from 'hooks/useOrderedConnections'
 import { ReactNode, useEffect, useMemo } from 'react'
+import { updateConnectionError } from 'state/connection/reducer'
+import { useAppDispatch } from 'state/hooks'
 
 export default function Web3Provider({ children }: { children: ReactNode }) {
+  const dispatch = useAppDispatch()
+
+  // Set metamask error handler for metamask disconnection warning modal.
+  useEffect(() => {
+    setErrorHandler((error: Error) =>
+      dispatch(updateConnectionError({ connectionType: ConnectionType.INJECTED, error: error.message }))
```

Comment:
Once this error is set, will it ever be unset?

Reply:
it gets unset on line 230 of src/components/WalletModal/index.tsx, but also the user is being prompted to reload the app with the metamask error, so it will get unset after the reload as well.